enhancements/olm/simplify-olm-apis.md

```diff
@@ -0,0 +1,366 @@
+
+- `matchLabels`: label selector used to list operator components
+- `components`: references to resources selected by `matchLabels`
+
+Add OLM controller logic such that, for each `Operator` resource:
+
+- `matchLabels` is resolved from the label convention and stored in the `status.resources.matchLabels` field
+
+```yaml
+status:
+  resources:
+    matchLabels:
+      operators.coreos.com/v2/operators/plumbus: ""
+```
+
+Add OLM controller logic such that, for each `Operator` resource:
+
+- cluster-scoped resources are selected using `status.resources.matchLabels`
+- namespace-scoped resources are selected __across all namespaces__ using `status.resources.matchLabels`
+- the union of both selections is stored in `status.resources.components`
+- kinds selected in both cases may be predefined or queried from discovery
+
+```yaml
+status:
+  resources:
+    matchLabels:
+      operators.coreos.com/v2/operator-name: plumbus
+    components:
+    - kind: Subscription
+      namespace: my-ns
+      name: plumbus
+      uid: d655a13e-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109719
+    - kind: ClusterServiceVersion
+      namespace: my-ns
+      name: plumbus.v2.0.0-alpha
+      uid: d70a53b5-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109811
+    # ...
+```
+
+Add OLM controller logic, such that for each `Operator` resource
+
+- OLM resources matching `status.resources.matchLabels` have the matching labels projected onto all related generated resources
+  - from `Subscriptions` onto resolved `Subscriptions` and `InstallPlans`
+  - from `InstallPlans` onto `ClusterServiceVersions`, `CustomResourceDefinitions`, RBAC, etc.
+  - from `ClusterServiceVersions` onto `Deployments` and `APIServices`
+
+Define, as part of the `Operator` resource, a `status.metadata` field that
+
+- contains the `displayName`, `description`, and `version` of an operator
+
+```yaml
+status:
+  metadata:
+    displayName: Plumbus
+    description: Welcome to the exciting world of Plumbus ownership! A Plumbus will aid many things in life, making life easier. With proper maintenance, handling, storage, and urging, Plumbus will provide you with a lifetime of better living and happiness.
+    version: 2.0.0-alpha
+```
+
+Define, as part of the `Operator` resource, a `status.apis` field that
+
+- contains the `required` and `provided` APIs of an operator
+
+```yaml
+status:
+  apis:
+    provides:
+    - group: how.theydoit.com
+      version: v2alpha1
+      kind: Plumbus
+      plural: plumbai
+      singular: plumbus
+    requires:
+    - group: how.theydoit.com
+      version: v1
+      kind: Grumbo
+      plural: grumbos
+      singular: grumbo
+```
+
+Define, as part of the `Operator` resource, a `status.permissions` field that
+
+- contains the RBAC requirements of an operator
+
+```yaml
+status:
+  cluster: # optional
```

Comment:
I don't think the rbac rules provide much value over just linking to the RBAC in `components`

Reply:
Also - if this does stay, the example should be `status.permissions.cluster`, not `status.cluster`, right?

Reply:
I'm dropping this since the RBAC should be sufficient.